### PR TITLE
add factory

### PR DESCRIPTION
**Pull request by twhiston**

### CODE REVIEW

File: bag.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ func New() *ParameterBag {
 
 }
 
+//ParameterBagConstructor can be registered with the  bag factory
+func ParameterBagConstructor() Bag {
```
Comment (daniele2387462786) [inline bag.go]: You are already in parameterbag. Maybe `parameterbag.ParameterBagConstructor` is a bit verbose? 
Maybe we can find a better naming

File: bagFactory.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package parameterbag
+
+import "github.com/pkg/errors"
```
Comment (daniele2387462786) [inline bagFactory.go]: I assume this is imported by accident?

**Review by daniele2387462786 — COMMENTED** [review]
I like where this is going

File: bag.go (hunk 1 above)
Comment (twhiston) [reply bag.go]: Changed to Construct

File: bagFactory.go (hunk 2 above)
Comment (twhiston) [reply bagFactory.go]: Yes it is. Fixed now

**Review by daniele2387462786 — APPROVED** [review]

**Pull request merged**